### ISSUE TITLE
Doubled file path

### ISSUE THREAD

**Issue by smfoote** [withfig/fig#455]
### Description:
Just updated my Mac OS version to 11.6.0 and started seeing this issue. When I select an option for a file path, that option is sort of doubled in the command I'm typing. The full option is entered, then the portion of the option that I had not yet typed is added after that.

https://user-images.githubusercontent.com/452458/133462625-fcb5ad2e-a673-4dff-ad2b-4bf59e7d58a3.mov

 
### Details:
|macOS|Fig|Shell|
|-|-|-|
|11.6.0|Version 1.0.48 (B264)|/bin/zsh|
<details><summary><code>fig diagnostic</code></summary>
<p>
<pre>Version 1.0.48 (B264) [U.S.] 
UserShell: /bin/zsh
Bundle path: /Applications/Fig.app
Autocomplete: true
Settings.json: true
CLI installed: true
CLI tool path: /Users/sfoote/.fig/bin/fig
Accessibility: true
Number of specs: 199
SSH Integration: false
Tmux Integration: true
Keybindings path: /Users/sfoote/.fig/user/keybindings
iTerm Integration: true 
Hyper Integration: true
VSCode Integration: true
Docker Integration: false
Symlinked dotfiles: false
Only insert on tab: false
Installation Script: true
PseudoTerminal Path: /Users/sfoote/.volta/bin:/Users/sfoote/.yarn/bin:/Users/sfoote/.config/yarn/global/node_modules/.bin:/Users/sfoote/.gem/ruby/X.X.0/bin:/Users/sfoote/.rbenv/shims:/usr/local/linkedin/bin:/usr/local/git/bin:/Users/sfoote/.nvm/versions/node/v10.13.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/export/content/linkedin/bin:/export/content/granular/bin:/Users/sfoote/.fig/bin:/Users/sfoote/Library/Android/sdk/platform-tools:/Users/sfoote/Library/Android/sdk/tools:/Users/sfoote/.rvm/bin:/export/content/linkedin/bin
SecureKeyboardInput: false
SecureKeyboardProcess: <none>
Current active process: ??? (???) - ???
Current working directory: ???
Current window identifier: ???</pre>
</p>
</details>

**Comment by smfoote** [comment]
UPDATE: it seems my terminal (Hyper) was having trouble properly loading zsh. I turned of Fig autocomplete and restarted the terminal and zsh loaded properly. Then, I turned Fig autocomplete back on and the issue is no longer happening. I will leave it up to the Fig team to decide whether or not to close the issue, but I am no longer blocked.

**Comment by mschrage** [comment]
Huh! Thanks for reporting this @smfoote! Glad the issue is resolved but would love to better understand what was going on.

Did this happen for any text that you inserted using Fig?

**Comment by smfoote** [comment]
I didn't try with too many commands, so I'm not sure the extent of the issue. I did see the insertion doubling with `git` commands (e.g. `git commitcommit`) as well as autocompleted file paths.

**Comment by ChenghaoMou** [comment]
Seeing the same issue with the terminal in VS Code. Tools like git, kubectl are also affected.
It is working fine in iTerm though.

**Comment by GuilleDF** [comment]
I'm having this issue as well on 

macOS  |	Fig  |	Shell
--------|-----|-----------
11.3.0     |	1.0.49    |	fish

**Comment by smfoote** [comment]
I saw this issue again today. The first two tabs I opened in Hyper (3.1.3) were working fine. The next tab had the issue.

**Comment by canac** [comment]
I'm seeing this (or a similar issue) with fish in iTerm 3.4.10. I can only reproduce it when there's multiple directories that match the substring. When I hit tab the first time, fish autocompletes the first part of the directories, but it looks like Fig isn't picking up that the input changed. Then when I hit tab again, it seems like Fig is completing again, leading to doubled text.

Reproduction:

- `mkdir test && cd test && mkdir foo1 && mkdir foo2`
- Type `cd f` (but don't hit `enter`)
- Hit `tab` (adds `oo`)
- Hit `tab` again (adds `oo` repeatedly)

https://user-images.githubusercontent.com/3740187/135138735-5376a170-7152-4f8f-b882-a6f014e44624.mov

|macOS|Fig|Shell|
|-|-|-|
|11.6.0|Version 1.0.49 (B270)|/usr/local/bin/fish|

**Comment by mschrage** [comment]
Hey @canac! Thanks so much for finding a way to reproduce this (same behavior on my machine when I am using `fish`)!

**Comment by mschrage** [comment]
@smfoote @canac @GuilleDF @ChenghaoMou I will try to figure out what's going on this week! If you find any other ways to reproduce the behavior or narrow down the environments where it occurs, please let me know!

**Comment by canac** [comment]
@mschrage Awesome! Just in case it's helpful, I noticed that my issue doesn't occur if the "Tab Behavior" setting is "insert". It does occur with all other values for that setting.

**Comment by mschrage** [comment]
@canac I think what's happening in the `fish` case is that when is in inserted, the autocomplete-engine's internal representation of the edit buffer is not updated correspondingly.

**Comment by GuilleDF** [comment]
Hi @mschrage, I did some testing and it seems to be related to the fact that I'm using hyper
Here's a repro video in case it's helpful

**Comment by mschrage** [comment]
@canac yeah, I was on a call yesterday with someone else who was running into this and it seems like this issue is specific to Hyper. (I've added a tag to reflect this)

**Comment by canac** [comment]
@mschrage OK, my issue definitely happens in iTerm (as well as Hyper). Once you fix that Hyper bug, I'll see if that fixes mine also.

**Comment by canac** [comment]
@mschrage Yeah, I can't reproduce my issue anymore with fish in iTerm. Great work! 🎉

**Comment by ChenghaoMou** [comment]
I can confirm it works on

- iTerm 2 + bash
- iTerm 2 + zsh
- iTerm 2 + zsh + oh-my-zsh
- iTerm 2 + bash + starship

on VS Code, where I used to have this problem:
- bash ❌  
- zsh ❌ 
- zsh + oh-my-zsh ✔️  
- bash + starship ❌ 

But now the problem I have within the VS Code is that the autocomplete doesn't seem to work in most cases:

![zsh](https://user-images.githubusercontent.com/32014649/138011361-df71eb25-f345-4459-b703-b227588ec7b8.png)

![bash](https://user-images.githubusercontent.com/32014649/138011354-319354c7-af05-4a6d-9735-2068f59e1411.png)

**Comment by GuilleDF** [comment]
@mschrage 
Still the same for me on hyper + fish. 😢 
In iTerm it was already working before for me, so no change there
